File: ios/Runner/SuperGpsStreamHandler.swift
import Flutter

final class SuperGpsStreamHandler: NSObject, FlutterStreamHandler {
    private var eventSink: FlutterEventSink?

    func onListen(
        withArguments arguments: Any?,
        eventSink events: @escaping FlutterEventSink
    ) -> FlutterError? {
        eventSink = events
        let options = SuperGpsStreamOptions.from(arguments: arguments)
        SuperGpsLocationEngine.shared.addListener(sink: events, options: options)
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        SuperGpsLocationEngine.shared.removeAllListeners()
        eventSink = nil
        return nil
    }
}

File: android/app/src/main/kotlin/com/sps/patrol/gps/SuperGpsStreamHandler.kt
package com.sps.patrol.gps

import android.content.Context
import io.flutter.plugin.common.EventChannel

class SuperGpsStreamHandler(private val appContext: Context) : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        if (events == null) return
        // onListen có thể gọi lại trước onCancel — gỡ sink cũ để không rò listener.
        eventSink?.let { SuperGpsLocationEngine.removeListener(it) }
        eventSink = events
        val options = SuperGpsStreamOptions.fromArguments(arguments)
        SuperGpsLocationEngine.addListener(appContext, events, options)
    }

    override fun onCancel(arguments: Any?) {
        eventSink?.let { SuperGpsLocationEngine.removeListener(it) }
        eventSink = null
    }
}
